c/g6-read-iterator.c

```diff
@@ -676,28 +705,51 @@ int _ReadGraphFromG6String(graphP pGraphToRead, char *g6EncodedString)
 		return NOTOK;
 	}
 
-	FILE *tmpG6Infile = tmpfile();
+	G6ReadIterator *pG6ReadIterator = NULL;
+	exitCode = allocateG6ReadIterator(&pG6ReadIterator, pGraphToRead);
 
-	if (tmpG6Infile == NULL)
+	if (exitCode != OK)
 	{
-		ErrorMessage("Unable to create temporary file to contain .g6 string contents.\n");
-		return NOTOK;
+		ErrorMessage("Unable to allocate G6ReadIterator.\n");
+		return exitCode;
 	}
 
-	int fputsCode = fputs(g6EncodedString, tmpG6Infile);
-	if (fputsCode == EOF)
+	strOrFileP g6Input = sf_New(NULL, g6EncodedString);
```

Comment:
I like the different beginG6ReadIteration methods with different type-based entry points. In C++, they could all have the same name and just accept different second parameters. In C, we use different names to compensate for absence of the function signature feature. Please add another method called `beginG6ReadIterationFromString()` to receive a char * as second parameter for a literal string. In its implemention, do the `sf_New()` code here then invoke `_beginG6ReadIteration()`. Then, at this point, invoke `beginG6ReadIterationFromString()`. This will do two things. First, you will have better consistency with the look of code between `_ReadGraphFromG6String()` and `_ReadGraphFromG6File()`. Second, the `sf_New()` will be at the same abstraction level as the `sf_Free()` (`endG6ReadIteration()` calls `sf_Free()` so a `beginG6ReadIteration ...  ()` method should be the thing that creates the string or file object).